Sources/Common/Store/SdkConfig.swift

```diff
@@ -75,3 +75,60 @@ public struct SdkConfig {
      */
     public var _sdkWrapperConfig: SdkWrapperConfig? // swiftlint:disable:this identifier_name
 }
+
+/**
+ We have a separate SDK config just for rich push because:
+ 1. Instance of SDK inside of a Notification Service Extension does not have as many features to provide
+    compared to running in a host app. Therefore, we don't need to expose as many SDK config options to customers.
+ 2. The SDK code needs to override some configuration options when running inside of a Notication Service Extension.
+    We don't want customers to modify some of these overriden config options as it may effect some features of rich push.
+
+ Note: To not make the SDK code more complex, convert `RichPushSdkConfig` to an instance of `SdkConfig` when SDK is initialized.
+ The SDK should not have conditional logic handling different SDK config objects. The SDK should only have to handle `SdkConfig`.
+ */
+/// SDK configuration just for rich push feature of the SDK.
+public struct RichPushSdkConfig {
```

Comment:
I believe, adding a use case code snippet would help !

Reply:
I added some comments to the class. See if those added comments answer your question.